apps/sms/views/shared/js/system_message_handler.js

```diff
@@ -270,11 +270,14 @@
 
         //Validate if message still exists before opening message thread
         //See issue https://bugzilla.mozilla.org/show_bug.cgi?id=837029
-        return Utils.onceDocumentIsVisible().then(
-          () => MessageManager.getMessage(id)
-        ).then(
+        return MessageManager.getMessage(id).then(
           (message) => Navigation.toPanel('thread', { id: message.threadId }),
-          () => Utils.alert('deleted-sms')
+          () => {
+            if (Navigation.hasPendingInit()) {
```

Comment:
I don't think the thread removed case is common, I thought user rarely have the chance to see the notification still exist after thread removed. But I still init the navigation here if it didn't finish the initialization with pending notification message case.